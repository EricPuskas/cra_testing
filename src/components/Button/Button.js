import React from "react";
import PropTypes from "prop-types";

const Button = ({ buttonText, emitEvent }) => {
  const handleSubmit = () => {
    if (emitEvent) emitEvent();
  };
  return (
    <button onClick={handleSubmit} data-test="SharedBtn">
      {buttonText}
    </button>
  );
};

Button.propTypes = {
  buttonText: PropTypes.string,
  emitEvent: PropTypes.func
};

export default Button;
