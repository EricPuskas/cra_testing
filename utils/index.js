import checkPropTypes from "check-prop-types";
import { applyMiddleware, createStore } from "redux";
import rootReducer from "../src/reducers";
import { middleware } from "../src/store";

export const findByAttribute = (component, attr) =>
  component.find(`[data-test="${attr}"]`);

export const checkProps = (component, expectedProps) =>
  checkPropTypes(component.propTypes, expectedProps, "props", component.name);

export const testStore = initState => {
  const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
  return createStoreWithMiddleware(rootReducer, initState);
};
