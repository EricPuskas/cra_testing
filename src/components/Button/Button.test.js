import React from "react";
import { shallow } from "enzyme";
import { findByAttribute, checkProps } from "../../../utils";
import Button from "./Button";

describe("Button Component", () => {
  describe("Checking PropTypes", () => {
    it("Should not throw a warning", () => {
      const expectedProps = {
        buttonText: "Example Button Text",
        emitEvent: () => {}
      };
      const propsError = checkProps(Button, expectedProps);
      expect(propsError).toBeUndefined();
    });
  });
});

describe("Renders without crashing", () => {
  let component;
  beforeEach(() => {
    const props = {
      buttonText: "Example Button Text",
      emitEvent: () => {}
    };
    component = shallow(<Button {...props} />);
  });
  it("Should pass snapshot test", () => {
    expect(component.debug()).toMatchSnapshot();
  });

  it("Should render a button", () => {
    const button = findByAttribute(component, "SharedBtn");
    expect(button.length).toBe(1);
  });
});
