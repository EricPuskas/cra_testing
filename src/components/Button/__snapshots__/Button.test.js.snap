// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing Should pass snapshot test 1`] = `
"<button onClick={[Function: handleSubmit]} data-test=\\"SharedBtn\\">
  Example Button Text
</button>"
`;
